geodistpy/distance.py

```
@@ -137,6 +143,352 @@ def geodist(coords1, coords2, metric="meter"):
+        # Coincident points
+        return [point1] * n_points
+    else:
+        total_dist = result[0]
+        fwd_az = result[1]
+
+    segment = total_dist / (n_points + 1)
+    waypoints = []
+    for i in range(1, n_points + 1):
+        lat, lon = geodesic_vincenty_direct(point1, fwd_az, segment * i)
+        if math.isnan(lat):
+            # Vincenty direct failed to converge – fall back to geographiclib
```

Ruff (0.15.1)

[warning] Comment contains ambiguous `–` (EN DASH). Did you mean `-` (HYPHEN-MINUS)?

(RUF003)